File: java/currency calculator/currencyCalc/app/src/main/java/com/gradproject/form/MainActivity.java
package com.gradproject.form;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;



public class MainActivity extends AppCompatActivity {

    public void convertToEGP(View view){
        EditText editText = (EditText) findViewById(R.id.USD1);
        TextView txt = findViewById(R.id.textView);
        double dollars = Double.parseDouble(editText.getText().toString());
        double egp = 15.7;

        double result = dollars * egp ;

        txt.setText(dollars+" USD equal "+result+" EGP");
    }
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

}

File: Kotlin dataset apps/Currency/MainActivity.kt
package com.badawy4.app4

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        button1.setOnClickListener({

            var USD:Double = USD1.text.toString().toDouble()
            val TOTAL = USD*15.7

            textView.text = " "+USD+" USD equal " + TOTAL + " EGP"

        }
        )

    }
}